### PR TITLE
NuSVM Docs

### PR DESCRIPTION
**Pull request by owerbat**

### CODE REVIEW

File: docs/source/onedal/algorithms/svm/svm.rst [hunk 1]
```diff
@@ -85,6 +93,38 @@ for reduced to the solution of the quadratic optimization problem
...
+    :math:`e^T\alpha \geq \nu`, :math:`y^T\alpha = 0`,
+    where :math:`e` is the vector of ones, :math:`\nu` is an upper bound on the fraction
+    of training errors and a lower bound of the fraction of support vector,
+    :math:`Q` is a symmetric matrix of size :math:`n \times n` with :math:`Q_{ij} = y_i y_j K(x_i, x_j)`,
+    and :math:`K(x,y)` is a kernel function.
+
+  .. group-tab:: Nu-regression
+
+    .. math::
+          \underset{\alpha }{\mathrm{min}}\frac{1}{2}{\alpha }^{T}Q\alpha
```
Comment (PetrovKP) [inline docs/source/onedal/algorithms/svm/svm.rst:111]: not correct formula 
image

File: docs/source/onedal/algorithms/svm/svm.rst [hunk 2]
```diff
@@ -42,13 +42,22 @@ and :math:`n` responses :math:`Y=\{y_1,\ldots,y_n\}`,
...
+    - :math:`y_i \in \{0, \ldots, M-1\}`, where :math:`M` is the number of classes
+
+  .. group-tab:: Nu-regression
+
+    - :math:`y_i \in \mathbb{R}`
+
+the problem is to build a Support Vector Machine (SVM) classification, regression,
+nu-classification, or nu-regression model.
 
 The SVM model is trained using the Sequential minimal optimization (SMO) method [Boser92]_}
```
Comment (PetrovKP) [inline docs/source/onedal/algorithms/svm/svm.rst:62]:
```suggestion
The SVM model is trained using the Sequential minimal optimization (SMO) method [Boser92]_
```

File: docs/source/onedal/algorithms/svm/svm.rst [hunk 3]
```diff
@@ -79,12 +88,45 @@ for reduced to the solution of the quadratic optimization problem
...
+         z_i = +1, s_i = \epsilon - y_i, & 0 < i \leq n \\
          z_i = -1, s_i = \epsilon + y_i, & n < i \leq 2n
       \end{cases}
 
     Where :math:`\epsilon` is the error tolerance parameter.
 
+  .. group-tab:: Nu-classification
+
+    .. math::
+          \underset{\alpha }{\mathrm{min}}\frac{1}{2}{\alpha }^{T}Q\alpha
```
Comment (PetrovKP) [inline docs/source/onedal/algorithms/svm/svm.rst:100]:
```suggestion
          \underset{\alpha }{\mathrm{min}} - \frac{1}{2}{\alpha }^{T}Q\alpha
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 4]
```diff
@@ -457,7 +463,8 @@ class model : public base {
     }
 
     /// The second unique value in class labels.
-    /// Used with :expr:`task::classification` only.
+    /// Used with :expr:`task::classification` and
+    /// :expr:`task::nu_classification` only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:467]:
```suggestion
    /// :expr:`task::nu_classification`.
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 5]
```diff
@@ -447,7 +452,8 @@ class model : public base {
     }
 
     /// The first unique value in class labels.
-    /// Used with :expr:`task::classification` only.
+    /// Used with :expr:`task::classification` and
+    /// :expr:`task::nu_classification` only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:456]:
```suggestion
    /// :expr:`task::nu_classification`.
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 6]
```diff
@@ -416,7 +419,9 @@ class model : public base {
     }
 
     /// A $nsv \\times class_count - 1$ table for :expr:`task::classification`
+    /// and :expr:`task::nu_classification`
     /// and $nsv \\times 1$ table for :expr:`task::regression`
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:423]:
```suggestion
    /// and a $nsv \\times 1$ table for :expr:`task::regression`
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 7]
```diff
@@ -342,7 +343,8 @@ class descriptor : public detail::descriptor_base<Task> {
     }
 
     template <typename T = Task, typename = detail::enable_if_classification_t<T>>
-    /// The number of classes. Used with :expr:`task::classification` only.
+    /// The number of classes. Used with :expr:`task::classification`
+    /// and :expr:`task::nu_classification` only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/common.hpp:347]:
```suggestion
    /// and :expr:`task::nu_classification`.
```

File: cpp/oneapi/dal/algo/svm/infer_types.hpp [hunk 8]
```diff
@@ -91,31 +97,34 @@ class infer_result : public base {
...
-    auto& set_labels(const table& value) {
+    auto &set_labels(const table &value) {
         set_labels_impl(value);
         return *this;
     }
 
-    // TODO: Enable this function only for classification problem.
-    // This will break binary compatibility, so the change should happen in 2022.1.
+    // TODO: Enable this function only for classification and nu_classification
+    // problem.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/infer_types.hpp:108]:
```suggestion
    // problems.
```

File: cpp/oneapi/dal/algo/svm/infer_types.hpp [hunk 9]
```diff
@@ -91,31 +97,34 @@ class infer_result : public base {
...
+    // This will break binary compatibility, so the change should happen in
+    // 2022.1.
     //
     /// The $n \\times 1$ table with the predicted class.
-    /// Used with :expr:`oneapi::dal::svm::task::classification` only.
+    /// Used with :expr:`oneapi::dal::svm::task::classification`
+    /// and :expr:`oneapi::dal::svm::task::nu_classification` only.
```
Comment (outoftardis) [inline cpp/oneapi/dal/algo/svm/infer_types.hpp:114]:
```suggestion
    /// and :expr:`oneapi::dal::svm::task::nu_classification`.
```

File: docs/source/onedal/algorithms/svm/svm.rst [hunk 10]
```diff
@@ -79,12 +88,45 @@ for reduced to the solution of the quadratic optimization problem
...
+
+    with :math:`0 \leq \alpha_i \leq 1`, :math:`i = 1, \ldots, n`,
+    :math:`e^T\alpha = n\nu`, :math:`y^T\alpha = 0`,
+    where :math:`e` is the vector of ones, :math:`\nu` is an upper bound on the fraction
+    of training errors and a lower bound of the fraction of support vector,
```
Comment (outoftardis) [inline docs/source/onedal/algorithms/svm/svm.rst:105]:
```suggestion
    of training errors and a lower bound of the fraction of the support vector,
```

File: docs/source/onedal/algorithms/svm/svm.rst [hunk 11]
```diff
@@ -79,12 +88,45 @@ for reduced to the solution of the quadratic optimization problem
...
+  .. group-tab:: Nu-regression
+
+    .. math::
+          \underset{\alpha }{\mathrm{min}}\frac{1}{2}{\alpha }^{T}Q\alpha + z^T\alpha
+
+    with :math:`0 \leq \alpha_i \leq \frac{C}{n}`, :math:`i = 1, \ldots, 2n`,
+    :math:`\sum_{i=1}^{n} \alpha_i - \sum_{i=n+1}^{2n} \alpha_i = 0`,
+    :math:`\sum_{i=1}^{2n} \alpha_i = C\nu`, where :math:`C` is the upper bound of the coordinates
+    of the vector :math:`\alpha`, :math:`\nu` is an upper bound on the fraction
+    of training errors and a lower bound of the fraction of support vector,
```
Comment (outoftardis) [inline docs/source/onedal/algorithms/svm/svm.rst:118]:
```suggestion
    of training errors and a lower bound of the fraction of the support vector,
```

File: docs/source/onedal/glossary.rst [hunk 12]
```diff
@@ -116,6 +116,16 @@ Machine learning terms
         **Example:** in linear regression algorithm, the model contains weight
         values for each input feature and a single bias value
 
+    Nu-classification
+        An SVM-specific :capterm:`classification` problem where :math:`\nu` parameter is used
+        instead of :math:`C`. :math:`\nu` is an upper bound on the fraction
+        of training errors and a lower bound of the fraction of support vector.
```
Comment (outoftardis) [inline docs/source/onedal/glossary.rst:122]:
```suggestion
        of training errors and a lower bound of the fraction of the support vector.
```

File: docs/source/onedal/glossary.rst [hunk 13]
```diff
@@ -116,6 +116,16 @@ Machine learning terms
...
+
+    Nu-regression
+        An SVM-specific :capterm:`regression` problem where :math:`\nu` parameter is used
+        instead of :math:`\epsilon`. :math:`\nu` is an upper bound on the fraction
+        of training errors and a lower bound of the fraction of support vector.
```
Comment (outoftardis) [inline docs/source/onedal/glossary.rst:127]:
```suggestion
        of training errors and a lower bound of the fraction of the support vector.
```

File: cpp/oneapi/dal/algo/svm/common.hpp [hunk 14]
```diff
@@ -436,9 +441,9 @@ class model : public base {
     }
 
     /// A $class_count*(class_count-1)/2 \\times 1$ table for
-    /// :expr:`task::classification`
-    /// and $1 \\times 1$ table for :expr:`task::regression`
-    /// calastable constants in decision function
+    /// :expr:`task::classification` and :expr:`task::nu_classification`
+    /// and $1 \\times 1$ table for :expr:`task::regression` and
+    /// :expr:`task::nu_regression` calastable constants in decision function
```
[the remark this answers is not in the document]
Comment (owerbat) [reply cpp/oneapi/dal/algo/svm/common.hpp]: Yes, it was just a typo

**Review by PetrovKP — APPROVED** [review]

**Review by outoftardis — APPROVED** [review]

**Comment by outoftardis** [comment]
@owerbat please also create a PR with these changes for the release branch

**Comment by PetrovKP** [comment]
@Mergifyio backport rls/2021.3-rls

**Pull request merged**